<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GLSLCodeEditorWidget</class>
 <widget class="QWidget" name="GLSLCodeEditorWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>682</width>
    <height>668</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>GLSL Fragment Shader Code Editor</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QToolButton { 
     border-style: solid;
     border-width: 1px;
     border-radius:5px;
	 padding: 3px;
     min-height: 24px;
     min-width: 24px;
     border-color: transparent;
     background: transparent;
}

QToolButton:checked { 
     border-color: palette(dark);
     background-color: qradialgradient(cx: 0.7, cy: 1.4, fx: 0.7, fy: 1.9, radius: 1.5, stop: 0 palette(dark), stop: 1 transparent );
}

QToolButton:pressed { 
	background: palette(midlight);
}

QToolButton:hover { 
     border-color: palette(highlight);
}

</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="childrenCollapsible">
      <bool>false</bool>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QTabWidget" name="PluginInformation">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>153</height>
          </size>
         </property>
         <property name="currentIndex">
          <number>1</number>
         </property>
         <widget class="QWidget" name="plugininfo">
          <attribute name="title">
           <string>Informations</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayouts">
           <item row="0" column="0" rowspan="3">
            <widget class="QLabel" name="label_5">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="pixmap">
              <pixmap resource="../icons.qrc">:/glmixer/icons/shadertoy_white.png</pixmap>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Name</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QLineEdit" name="nameEdit"/>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>About</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QLineEdit" name="aboutEdit"/>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Description</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QLineEdit" name="descriptionEdit"/>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="shaderinput">
          <attribute name="title">
           <string>Shader input</string>
          </attribute>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <property name="spacing">
            <number>0</number>
           </property>
           <property name="leftMargin">
            <number>3</number>
           </property>
           <property name="topMargin">
            <number>3</number>
           </property>
           <property name="rightMargin">
            <number>3</number>
           </property>
           <property name="bottomMargin">
            <number>3</number>
           </property>
           <item>
            <widget class="GLSLCodeEditor" name="headerText">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>114</height>
              </size>
             </property>
             <property name="undoRedoEnabled">
              <bool>false</bool>
             </property>
             <property name="html">
              <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Ubuntu'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="textInteractionFlags">
              <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item>
        <widget class="GLSLCodeEditor" name="codeTextEdit">
         <property name="html">
          <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Ubuntu'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QToolButton" name="loadButton">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="../icons.qrc">
             <normaloff>:/glmixer/icons/folderopen.png</normaloff>:/glmixer/icons/folderopen.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>24</width>
             <height>24</height>
            </size>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="pasteButton">
           <property name="text">
            <string>Paste over</string>
           </property>
           <property name="icon">
            <iconset resource="../icons.qrc">
             <normaloff>:/glmixer/icons/textpaste.png</normaloff>:/glmixer/icons/textpaste.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>24</width>
             <height>24</height>
            </size>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="applyButton">
           <property name="text">
            <string>Apply</string>
           </property>
           <property name="icon">
            <iconset resource="../icons.qrc">
             <normaloff>:/glmixer/icons/media-playback-start.png</normaloff>:/glmixer/icons/media-playback-start.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>24</width>
             <height>24</height>
            </size>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="validateButton">
           <property name="text">
            <string>Apply &amp;&amp; Close</string>
           </property>
           <property name="icon">
            <iconset resource="../icons.qrc">
             <normaloff>:/glmixer/icons/media-playback-start.png</normaloff>:/glmixer/icons/media-playback-start.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>24</width>
             <height>24</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="label">
         <property name="styleSheet">
          <string notr="true">background: palette(dark);</string>
         </property>
         <property name="text">
          <string>GLSL Compilation logs</string>
         </property>
         <property name="margin">
          <number>3</number>
         </property>
         <property name="indent">
          <number>15</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget2">
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="sizeConstraint">
        <enum>QLayout::SetNoConstraint</enum>
       </property>
       <item>
        <widget class="QPlainTextEdit" name="logText">
         <property name="acceptDrops">
          <bool>false</bool>
         </property>
         <property name="undoRedoEnabled">
          <bool>false</bool>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="plainText">
          <string notr="true"/>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLSLCodeEditor</class>
   <extends>QTextEdit</extends>
   <header location="global">GLSLCodeEditor.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>applyButton</sender>
   <signal>pressed()</signal>
   <receiver>GLSLCodeEditorWidget</receiver>
   <slot>apply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>661</x>
     <y>483</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>197</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>validateButton</sender>
   <signal>pressed()</signal>
   <receiver>GLSLCodeEditorWidget</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>603</x>
     <y>403</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>validateButton</sender>
   <signal>pressed()</signal>
   <receiver>GLSLCodeEditorWidget</receiver>
   <slot>apply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>603</x>
     <y>403</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>apply()</slot>
 </slots>
</ui>
